src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int RESIZE_INDEX = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private Node<K,V>[] elements;
+    private int size;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int hashValue = hash(key);
+        int index = getIndex(key, capacity);
+        for (Node<K,V> node = elements[index]; node != null; node = node.next) {
+            if (node.hash == hashValue && Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+        }
+        putValue(key, value, hashValue,index);
+    }
 
+    public boolean isEmpty() {
+        return size == 0;
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        int index = getIndex(key, capacity);
+        for (Node<K, V> node = elements[index]; node != null; node = node.next) {
+            if (Objects.equals(node.key, key) && node.hash == hash(key)) {
```

Comment:
if objects are equal, their hashcodes are equal too

Why do you compare hashs, if you already compare keys?

Reply:
I agree) Let's use only one of them